test/active_record_doctor/detectors/mismatched_foreign_key_type_test.rb

```diff
@@ -15,6 +15,22 @@ def test_mismatched_foreign_key_type_is_reported
     OUTPUT
   end
 
+  def test_matched_foreign_key_with_non_primary_key_type_is_not_reported
+    # MySQL does not allow foreign keys to have different type than paired primary keys
+    return if mysql?
+
+    create_table(:companies, id: :bigint) do |t|
+      t.integer :entity_id
+      t.index [:entity_id], unique: true
```

Comment:
I think this can be simplified to just:
```ruby
t.integer :entity_id, index: true
```

There is no need for it to be `unique`.